src/ast/ddl.rs

```diff
@@ -157,6 +157,28 @@ pub enum AlterTableOperation {
     SwapWith { table_name: ObjectName },
     /// 'SET TBLPROPERTIES ( { property_key [ = ] property_val } [, ...] )'
     SetTblProperties { table_properties: Vec<SqlOption> },
+
+    /// `OWNER TO { <new_owner> | CURRENT_ROLE | CURRENT_USER | SESSION_USER }`
+    ///
+    /// Note: this is a PostgreSQL-specific operation.
+    OwnerTo { new_owner: Owner },
+}
+
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub enum Owner {
+    Ident(Ident),
+    Expr(Expr),
+}
```

Comment:
```suggestion
pub enum Owner {
    Ident(Ident),
    CurrentRole,
    CurrentUser
    SessionUser
}
```
I'm thinking the Expr might be somewhat opaque to represent the others, from the docs the other variants seem enumerable so figured we can represent them verbatim in the enum as above?

Reply:
@iffyio Thanks for the review!

Currently `CURRENT_USER`, and `SESSION_USER` are supported as Expr::Function like this.

```
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] Parsing sql 'SELECT CURRENT_USER, CURRENT_ROLE, SESSION_USER;
'...
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] parsing expr
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] prefix: Function(Function { name: ObjectName([Ident { value: "CURRENT_USER", quote_style: None }]), args: None, filter: None, null_treatment: None, over: None, within_group: [] })
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] get_next_precedence() TokenWithLocation { token: Comma, location: Location { line: 1, column: 20 } }
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] 0: , 1: CURRENT_ROLE 2: ,
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] next precedence: 0
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] parsing expr
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] prefix: Identifier(Ident { value: "CURRENT_ROLE", quote_style: None })
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] get_next_precedence() TokenWithLocation { token: Comma, location: Location { line: 1, column: 34 } }
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] 0: , 1: SESSION_USER 2: ;
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] next precedence: 0
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] parsing expr
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] prefix: Function(Function { name: ObjectName([Ident { value: "SESSION_USER", quote_style: None }]), args: None, filter: None, null_treatment: None, over: None, within_group: [] })
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] get_next_precedence() TokenWithLocation { token: SemiColon, location: Location { line: 1, column: 48 } }
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] 0: ; 1: EOF 2: EOF
2024-06-22T13:04:10.639Z DEBUG [sqlparser::parser] next precedence: 0
```
https://github.com/sqlparser-rs/sqlparser-rs/blob/f16c1afed0fa273228e74a633f3885c9c6609911/src/parser/mod.rs#L998-L1010

I think it would be better to maintain this compatibility.

Additionally, `CURRENT_ROLE` must also be considered.

Reply:
Ah I see, I think it makes a lot of sense to represent them as functions when parsing arbitrary expressions - when parsing an `Owner` however, spontaneously doesn't feel the same since the context is much more restricted. Thinking here we can be explicit about what comes out of the parser which would be desirable from an API pov.

Reply:
I agree it is better to be explicit and match the allowed postgres syntax more exactly. I double checked https://www.postgresql.org/docs/current/sql-altertable.html and it says:

>     OWNER TO { new_owner | CURRENT_ROLE | CURRENT_USER | SESSION_USER }
>
> ...
> new_owner 
>   The user name of the new owner of the table.


So I think @iffyio 's suggestion is a good one.

I took the liberty of making this change in 0ff17d7 as I am preparing for a sqlparser release